### PR TITLE
DOC: Add DIPY dataset list to documentation.

### PR DESCRIPTION
**Pull request by jhlegarreta**
Add DIPY dataset list to documentation.

### CODE REVIEW

**Comment by jhlegarreta** [comment]
If I am not mistaken, having a list of the datasets available in DIPY was suggested as a helpful tool for users at some point during a DIPY lab meeting a while ago.

Please review and comment as appropriate. Thanks.

**Comment by arokem** [comment]
This is great! It would be good to add the relevant reference for each dataset.

**Comment by jhlegarreta** [comment]
@arokem it could be added in the `Citations` column, or the column header could be renamed to `References`.

I may not have all the relevant information though, since not all datasets' description in [fetcher.py](https://github.com/dipy/dipy/blob/master/dipy/data/fetcher.py) includes the source.

In any case, now I am thinking that, at the same time, we should avoid duplicating the information, so I am hesitating what part of the information should go where.

**Comment by arokem** [comment]
I think that it would be nice to have the references here. If only for
selfish reasons (the Stanford HARDI one is a dataset I collected and I know
that not everyone who uses it cites me for it). I think that we can
collectively figure out all the relevant references.

**Comment by jhlegarreta** [comment]
OK, sounds reasonable. I will try to figure out the references and will comment when I have difficulty in finding any. Thanks Ariel.

**Comment by jhlegarreta** [comment]
@arokem A few comments:

Could not locate the sources/references for the datasets:
- CENIR HCP-like dataset (`cenir_multib`)
- Gold standard streamlines IO testing (`gold_standard_io`)
- ISBI2013 (`isbi2013_2shell`)
- IVIM (`fetch_ivim`)
- qt-dMRI C57Bl6 mice datase (`qtdMRI_test_retest_2subject`)
- SNAIL (`bundles_2_subjects`)
- SyN (`syn_data`)
- Tissue data (`tissue_data`)
- Tractogram file formats examples (`bundle_file_formats_example`)

The following links seem not to be reachable:
- Tractogram file formats examples (`bundle_file_formats_example`)
https://zenodo.org/record/3352379/files/

- qt-dMRI C57Bl6 mice datase (`qtdMRI_test_retest_2subjects`)
https://zenodo.org/record/996889/files/

- Gold standard streamlines IO testing (`gold_standard_io`)
https://zenodo.org/record/2651349/files/

Not sure about the sources of the HCP datasets or how the bundles were generated.

- HCP842 bundle atlas (`bundle_atlas_hcp842`)
- HCP bundle FA (`bundle_fa_hcp`)
- HCP tractogram (`target_tractogram_hcp`)

Was the HCP-842 built by the DSI Studio folks, including the bundles: http://brain.labsolver.org/diffusion-mri-templates/hcp-842-hcp-1021?

The ISBI2013 (`isbi2013_2shell`) dataset is it the 2013 ISBI Challenge dataset?
http://hardi.epfl.ch/static/events/2013_ISBI/testing_data.html
The b-values in the data and the ones in the website do not match

As for the Stanford-named datasets, not sure if all come from the same reference.
Also, I assumed they come from here
https://purl.stanford.edu/ng782rw8378
But the fetcher URL seems not to exist
https://stacks.stanford.edu/file/druid:yx282xq2090

As for the NTU DSI data (`taiwan_ntu_dsi`), is the DSI studio the original site where the data was hosted, should we cross ref the DSI Studio website, the live box link?
http://brain.labsolver.org/diffusion-mri-data/ntu-dmri-data

Thanks.

**Comment by skoudoro** [comment]
Hi @jhlegarreta, 

I think you just need to remove the `/files/` to reach them.

> The following links seem not to be reachable:
> 
> Tractogram file formats examples (bundle_file_formats_example)
> https://zenodo.org/record/3352379/files/

reachable via https://zenodo.org/record/3352379 or I do not understand what do you mean

> qt-dMRI C57Bl6 mice datase (qtdMRI_test_retest_2subjects)
> https://zenodo.org/record/996889/files/

reachable via https://zenodo.org/record/996889 or I do not understand what do you mean

> Gold standard streamlines IO testing (gold_standard_io)
> https://zenodo.org/record/2651349/files/

reachable via https://zenodo.org/record/2651349 or I do not understand what do you mean

File: doc/data.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+.. include:: dataset_list.rst
+
+The list of datasets can be retrieved using::
+
+    from dipy.workflows.io import FetchFlow
+
+    available_data = FetchFlow.get_fetcher_datanames().keys()
+
+
+To retrieve all datasts, the following workflow can be run::
```
Comment (skoudoro) [inline doc/data.rst]: typo: `datasets`

File: doc/dataset_list.rst [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+        <td>scil_b0</td>
+        <td><a href='http://scil.dinf.usherbrooke.ca/fr/'>Sherbrooke Connectivity Imaging Lab (SCIL)</a></td>
+    </tr>
+    <tr>
+        <td>Sherbrooke 3 shells</td>
+        <td>Human</td>
+        <td>Multi shell data: b-vals: [0, 1000, 2000; 3500] (s/mm^2); 193 gradient directions</td>
+        <td></td>
+        <td>sherbrooke_3shell</td>
+        <td><a href='http://scil.dinf.usherbrooke.ca/fr/'>Sherbrooke Connectivity Imaging Lab (SCIL)</a></td>
```
Comment (skoudoro) [inline doc/dataset_list.rst]: I think it is better to remove `/fr/`. I suppose the redirection is done following your browser language.

**Comment by Garyfallidis** [comment]
> Could not locate the sources/references for the datasets:

What do you mean exactly? Are you talking about the online links? Or citations?
You cannot see the links from the fetchers? Did you look in fetcher.py?

> - CENIR HCP-like dataset (`cenir_multib`)
> - Gold standard streamlines IO testing (`gold_standard_io`)
> - ISBI2013 (`isbi2013_2shell`)
> - IVIM (`fetch_ivim`)
> - qt-dMRI C57Bl6 mice datase (`qtdMRI_test_retest_2subject`)
> - SNAIL (`bundles_2_subjects`)
> - SyN (`syn_data`)
> - Tissue data (`tissue_data`)
> - Tractogram file formats examples (`bundle_file_formats_example`)
> 
> The following links seem not to be reachable:
> - Tractogram file formats examples (`bundle_file_formats_example`)
> https://zenodo.org/record/3352379/files/
> 
> - qt-dMRI C57Bl6 mice datase (`qtdMRI_test_retest_2subjects`)
> https://zenodo.org/record/996889/files/
> 
> - Gold standard streamlines IO testing (`gold_standard_io`)
> https://zenodo.org/record/2651349/files/
> 
> Not sure about the sources of the HCP datasets or how the bundles were generated.
> 
> - HCP842 bundle atlas (`bundle_atlas_hcp842`)
This is not exactly the same with what is in DSI Studio. We have curated it by removing some unrealistic bundles and transformed it to MNI 152 (ICBM 2009a) space.
> - HCP bundle FA (`bundle_fa_hcp`)
> - HCP tractogram (`target_tractogram_hcp`)
These two are 
> 
> Was the HCP-842 built by the DSI Studio folks, including the bundles: http://brain.labsolver.org/diffusion-mri-templates/hcp-842-hcp-1021?

See above. We got this from there and then updated it. Our own version is in Figshare.

> The ISBI2013 (`isbi2013_2shell`) dataset is it the 2013 ISBI Challenge dataset?
> http://hardi.epfl.ch/static/events/2013_ISBI/testing_data.html
Correct
> The b-values in the data and the ones in the website do not match
Not sure why. Maybe there was an older version and a newer version.

> As for the Stanford-named datasets, not sure if all come from the same reference.
> Also, I assumed they come from here
> https://purl.stanford.edu/ng782rw8378
> But the fetcher URL seems not to exist
> https://stacks.stanford.edu/file/druid:yx282xq2090
> 
Okay @arokem can help with that.

> As for the NTU DSI data (`taiwan_ntu_dsi`), is the DSI studio the original site where the data was hosted, should we cross ref the DSI Studio website, the live box link?
> http://brain.labsolver.org/diffusion-mri-data/ntu-dmri-data
Okay. Please also check if b-values match.


> 
> Thanks.

File: doc/dataset_list.rst [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+        <td>qtdMRI_test_retest_2subject</td>
+        <td></td>
+    </tr>
+    <tr>
+        <td>SCIL b0</td>
+        <td></td>
+        <td>b0</td>
+        <td>GE (1.5, 3 T), Philips (3 T); Siemens (1.5, 3 T)</td>
+        <td>scil_b0</td>
+        <td><a href='http://scil.dinf.usherbrooke.ca/fr/'>Sherbrooke Connectivity Imaging Lab (SCIL)</a></td>
```
Comment (skoudoro) [inline doc/dataset_list.rst]: I think it is better to remove `/fr/`. I suppose the redirection is done following your browser language.

Comment (jhlegarreta) [reply doc/dataset_list.rst]: Done in 873ad5e. Thanks.

File: doc/dataset_list.rst (hunk 2 above)
Comment (jhlegarreta) [reply doc/dataset_list.rst]: Done in 873ad5e. Thanks.

**Comment by jhlegarreta** [comment]
> Thank you for this @jhlegarreta.
> 
> Below, some quick proposition to fix. Did you try to generate these files? Can you add a screenshot to see how it looks like? You can generate the doc without examples which is better and fast in this case 15-30sec.
> 
> Thank you

As discussed off-line, it looks like documentation does not get generated on my computer for some reason, so we will need to trust the CI builds if the docs are generated on any.

**Comment by jhlegarreta** [comment]
@skoudoro Thanks for your [comments](https://github.com/dipy/dipy/pull/2057#issuecomment-591696651). They allowed to me retrieve the references for those datasets which I added in 873ad5e.

**Comment by jhlegarreta** [comment]
@Garyfallidis Thanks for your [comments](https://github.com/dipy/dipy/pull/2057#issuecomment-591698761). I added the references for the 2013 ISBI Challenge dataset and the one that you hold in figshare.

I did not re-check the b-values for the data; I left the ones that I got the first time I queried the data, but there may well be different versions of the data and the sources/descriptions be incomplete/unclear (and hence the difference in the b-values, for example).

**Comment by jhlegarreta** [comment]
Folks, I think this is ready to be merged. The table contains complete information to the extent that it is currently available.

As said, I cannot build the documentation locally, but the table's HTML syntax is correct/free online tools are able to display it correctly.

@arokem Can you please confirm that the citation for all Stanford-labelled datasets is the same and matches the one I mentioned [here](https://github.com/dipy/dipy/pull/2057#issuecomment-590564101)?

@Garyfallidis, to wrap-up the follow-up comment [here](https://github.com/dipy/dipy/pull/2057#issuecomment-593591859), I re-checked the ISBI 2013 Challenge dataset's (`isbi2013_2shell`) `bval` file and it contains the `0`, `1500`, and `2500` s/mm^2 b-values, which do not match the challenge website's _current_ /_available_ information ([here](http://hardi.epfl.ch/static/events/2013_ISBI/training_data.html) and [here](http://hardi.epfl.ch/static/events/2013_ISBI/testing_data.html)). As suggested, maybe they correspond to data that was actually used within the context of the challenge but not used for the benchmarking. So this is fine to me unless we can cast more light on it.

**Comment by arokem** [comment]
> @arokem Can you please confirm that the citation for all Stanford-labelled datasets is the same and matches the one I mentioned here?

Yes. That's the right citation for the Stanford dataset and its derivatives.

**Comment by jhlegarreta** [comment]
@skoudoro this is ready to be merged from my end.

**Comment by skoudoro** [comment]
Thank you @jhlegarreta  for this. Merging

**Pull request merged**